lib/ansible/modules/cloud/amazon/polly_speech.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2018, Aaron Smith <ajsmith10381@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = r'''
+---
+module: polly_speech
+short_description: Synthesize speech using AWS Polly.
+description:
+    - Synthesizes UTF-8 input, plain text or SSML, to a stream of bytes. SSML input must be valid, well-formed SSML.
+author: "Aaron Smith (@slapula)"
+version_added: "2.7"
+requirements: [ 'botocore', 'boto3' ]
+options:
+  lexicons:
+    description:
+    - List of one or more pronunciation lexicon names you want the service to apply during synthesis.
+  output_format:
+    description:
+    - The format in which the returned output will be encoded.
+    choices: ['json', 'mp3', 'ogg_vorbis', 'pcm']
+    required: true
+  sample_rate:
+    description:
+    - The audio frequency specified in Hz.
+    default: '16000'
+  speech_mark_types:
+    description:
+    - The type of speech marks returned for the input text.
+    choices: ['sentence', 'ssml', 'viseme', 'word']
+  text_type:
+    description:
+    - Specifies whether the input text is plain text or SSML.
+    choices: ['ssml', 'text']
+    default: 'text'
+  voice_id:
+    description:
+    - Voice ID to use for the synthesis.
+    choices: ['Geraint', 'Gwyneth', 'Mads', 'Naja', 'Hans', 'Marlene', 'Nicole', 'Russell', 'Amy', 'Brian', 'Emma', 'Raveena', 'Ivy', 'Joanna',
+        'Joey', 'Justin', 'Kendra', 'Kimberly', 'Matthew', 'Salli', 'Conchita', 'Enrique', 'Miguel', 'Penelope', 'Chantal', 'Celine', 'Mathieu',
+        'Dora', 'Karl', 'Carla', 'Giorgio', 'Mizuki', 'Liv', 'Lotte', 'Ruben', 'Ewa', 'Jacek', 'Jan', 'Maja', 'Ricardo', 'Vitoria', 'Cristiano',
+        'Ines', 'Carmen', 'Maxim', 'Tatyana', 'Astrid', 'Filiz', 'Vicki', 'Takumi', 'Seoyeon', 'Aditi']
+    required: true
+  source:
+    description:
+    - Input text to synthesize.
+    - Specifies where to get the text file needed for processing.
+    - If you specify ssml as the `text_type`, follow the SSML format for the input text.
+    choices: ['local', 's3']
+    required: true
+  src_bucket:
+    description:
+    - Specifies the S3 bucket that stores the text file.
+  src_path:
+    description:
+    - Specifies the local path or S3 key path where the text file resides.
+  destination:
```

Comment:
Maybe instead an additional parameter to specify the destination, make dst_bucket and dst_path mutually exclusive and require one of them.
```
AnsibleAWSModule(...,
                 required_one_of=[('dst_bucket', 'dst_path')],
                 mutually_exclusive=[('dst_bucket', 'dst_path')])
```

Reply:
Maybe I should make this a little more clear in the documentation but both `src_path` and `dst_path` have shared responsibility.  When the source and/or destination are local, the `_path` parameter functions as the absolute or relative path to the text file on the filesystem.  When the source and/or destination are S3, `_path` parameter functions as the S3 key in relation to the bucket.  To make a long story short, `_path` will always be used but `_bucket` will only be needed when S3 is designated as the source or destination.